### ISSUE TITLE
DOT via GEMM TPP

### ISSUE THREAD

**Issue by alheinecke** [libxsmm/libxsmm#803]
For various LLM and GNN operators, we need vector a fast dot. Right now we have only very slow A^T GEMM for M=1 or we have to run a sequence of unary/binary TPPs.

Plan for improvement: Add fast A^T GEMM for M=1 which uses an inner-product approach + vector reduce in the end.

**Comment by hfp** [comment]
General Q (raised by others), this would not benefit from AMX as the latter needs matrices/tiles to benefit from reuse etc.  
Is this a correct assumption?

**Comment by egeor** [comment]
DOT and GEMV are BW BOUND ops, so AMX is irrelevant

**Comment by hfp** [comment]
> DOT and GEMV are BW BOUND ops, so AMX is irrelevant

Indeed. However, this came up in a context of "batched dot-products" which in turn can be thought like a matrix multiplication. I still wonder if there is something useful in it and if I should get the exact use case.

**Comment by egeor** [comment]
Well, then one is better off by reformulating the  algorithm/math to use
matmul. This is a standard trick in linear algebra…

**Comment by hfp** [comment]
For the record, this can be related to https://github.com/vondele/Stockfish/tree/amx_v1.
